src/ondil/distributions/powerexponential.py

```diff
@@ -0,0 +1,203 @@
+        mu, sigma, nu = self.theta_to_params(theta)
+        z = self._z(y, mu, sigma)
+        c = self._c(nu)
+
+        if param == 0:
+            return (np.sign(z) * nu / (2 * sigma * np.abs(z))) * (np.abs(z / c) ** nu)
+        elif param == 1:
+            return ((nu / 2) * (np.abs(z / c) ** nu) - 1) / sigma
+        elif param == 2:
+            logc_deriv = (1 / (2 * nu**2)) * (
+                2 * np.log(2) - digamma(1 / nu) + 3 * digamma(3 / nu)
+            )
+            s = np.abs(z / c) ** nu
+            return (
+                (1 / nu)
+                - 0.5 * np.log(np.abs(z / c)) * s
+                + np.log(2) / (nu**2)
+                + digamma(1 / nu) / (nu**2)
+                + (-1 + (nu / 2) * s) * logc_deriv
+            )
+
+    def d2ldv2(self, y, theta):
+        mu, sigma, nu = self.theta_to_params(theta)
+
+        dlogc_dv = (1 / (2 * nu**2)) * (
+            2 * np.log(2) - digamma(1 / nu) + 3 * digamma(3 / nu)
+        )
+
+        p = (1 + nu) / nu
+        part1 = p * polygamma(1, p) + 2 * digamma(p) ** 2
+        part2 = digamma(p) * (np.log(2) + 3 - 3 * digamma(3 / nu) - nu)
+        part3 = -3 * digamma(3 / nu) * (1 + np.log(2))
+        part4 = -(nu + np.log(2)) * np.log(2)
+        part5 = -nu + (nu ** 4) * (dlogc_dv ** 2)
+
+        d2ldv2 = -(part1 + part2 + part3 + part4 + part5) / (nu ** 3)
+        d2ldv2 = np.where(d2ldv2 < -1e-15, d2ldv2, -1e-15)
+        return d2ldv2
+
+    def dl2_dp2(self, y, theta, param):
+        self._validate_dln_dpn_inputs(y, theta, param)
+        mu, sigma, nu = self.theta_to_params(theta)
+
+        if param == 0:
+            z = self._z(y, mu, sigma)
+            c = self._c(nu)
+            dldm = (np.sign(z) * nu / (2 * sigma * np.abs(z))) * (np.abs(z / c) ** nu)
+            closed_form = -(nu ** 2 * gamma(2 - 1 / nu) * gamma(3 / nu)) / ((sigma * gamma(1 / nu)) ** 2)
+            d2ldm2 = np.where(nu < 1.05, -dldm ** 2, closed_form)
+            return d2ldm2
+
+        if param == 1:
+            return -nu / (sigma ** 2)
+
+        if param == 2:
+            return self.d2ldv2(y, theta)
+
+    def dl2_dpp(self, y, theta, params):
+        self._validate_dl2_dpp_inputs(y, theta, params)
+        if set(params) == {0, 1}:
+            return np.zeros_like(y)
+        return -self.dl1_dp1(y, theta, params[0]) * self.dl1_dp1(y, theta, params[1])
+
+    def initial_values(self, y: np.ndarray, param=None, axis=None) -> np.ndarray:
+
+        y = np.asarray(y)
+        mu_init = np.full_like(y, np.mean(y))
+        sigma_init = np.full_like(y, np.std(y, ddof=1))
+        nu_init = np.full_like(y, np.log(2))
+
+        if param is None:
+            return np.stack((mu_init, sigma_init, nu_init), axis=1)
+        elif param == 0:
+            return mu_init
+        elif param == 1:
+            return sigma_init
+        elif param == 2:
+            return nu_init
+        else:
+            raise ValueError("Invalid parameter index")
```

Comment:
Any particular reason why you chose to keep the `param` and `axis` parameters? If not please accept this change which removes those (and makes the `initial_values` implementation consistent with the other implementations).

```suggestion
    def initial_values(self, y: np.ndarray) -> np.ndarray:
        mu_init = np.full_like(y, np.mean(y))
        sigma_init = np.full_like(y, np.std(y, ddof=1))
        nu_init = np.full_like(y, np.log(2))
        return np.stack((mu_init, sigma_init, nu_init), axis=1)
```